### PR TITLE
Add support for Polygon/Matic

### PR DESCRIPTION
**Pull request by MysticRyuujin**
Is this all that is required? I think it is...

### CODE REVIEW

**Comment by splix** [comment]
Thank you! I'll test with this changes, fix the unit tests and merge

**Comment by AJolly** [comment]
That would be great!

File: src/main/kotlin/io/emeraldpay/dshackle/upstream/calls/DefaultEthereumMethods.kt [hunk 1]
```diff
@@ -119,6 +119,9 @@ class DefaultEthereumMethods(
                     Chain.ETHEREUM_CLASSIC == chain -> {
                         "\"1\""
                     }
+                    Chain.POLYGON == chain -> {
+                        "\"1\""
```
Comment (zhongfu) [inline src/main/kotlin/io/emeraldpay/dshackle/upstream/calls/DefaultEthereumMethods.kt:123]: I believe this should be `"\"137\""` instead:

```
> net.version
"137"
```

File: proto/common.proto [hunk 2]
```diff
@@ -15,6 +15,8 @@ enum ChainRef {
     CHAIN_ETHEREUM = 100;
     CHAIN_ETHEREUM_CLASSIC = 101;
 
+    CHAIN_POLYGON = 137;
```
Comment (zhongfu) [inline proto/common.proto:18]: [`proto/common.proto` in emeraldpay/emerald-grpc](https://github.com/emeraldpay/emerald-grpc/blob/master/proto/common.proto) seems to use id 1002 instead:

```
    // Sidechains and state channels start with 1_000
    [...]
    CHAIN_MATIC = 1002; // Matic PoS Ethereum sidechain based on Polygon
```

File: src/main/kotlin/io/emeraldpay/dshackle/upstream/calls/DefaultEthereumMethods.kt (hunk 1 above)
Comment (MysticRyuujin) [reply src/main/kotlin/io/emeraldpay/dshackle/upstream/calls/DefaultEthereumMethods.kt:123]: This is the result for eth_chainId, not net_version which is set elsewhere.

File: proto/common.proto (hunk 2 above)
Comment (MysticRyuujin) [reply proto/common.proto:18]: I assume this is the maintainer trying to implement support outside of this PR. This PR can be updated or closed once his work is complete.

File: src/main/kotlin/io/emeraldpay/dshackle/upstream/calls/DefaultEthereumMethods.kt (hunk 1 above)
Comment (MysticRyuujin) [reply src/main/kotlin/io/emeraldpay/dshackle/upstream/calls/DefaultEthereumMethods.kt:123]: Wait nevermind, you're right. I thought this was weird when I did it, and I remembered incorrectly. This is net_version and I do get 137 now

**Pull request merged**